libs/ledger-live-common/src/families/ethereum/hw-signMessage.ts

```diff
@@ -122,4 +134,149 @@ const signMessage: EthSignMessage = async (
   };
 };
 
-export default { prepareMessageToSign, signMessage };
+/**
+ * Get the value at a specific path of an object and return it as a string or as an array of string
+ * Used recursively by getValueFromPath
+ *
+ * @see getValueFromPath
+ */
+const getValue = (
+  path: string,
+  value: Record<string, any> | Array<any> | string
+): Record<string, any> | Array<any> | string => {
+  if (typeof value === "object") {
+    if (Array.isArray(value)) {
+      return value.map((v) => getValue(path, v)).flat();
+    }
+
+    if (!Object.prototype.hasOwnProperty.call(value, path)) {
+      throw new Error();
+    }
+    const result = value[path];
+    return typeof result === "object" ? result : result.toString();
+  }
+
+  return value.toString();
+};
+
+/**
+ * Using a path as a string, returns the value(s) of a json key without worrying about depth or arrays
+ * (e.g: 'to.wallets.[]' => ["0x123", "0x456"])
+ */
+const getValueFromPath = (
+  path: string,
+  eip721Message: EIP712Message
+): string | string[] => {
+  const splittedPath = path.split(".");
+  const { message } = eip721Message;
+
+  let value: any = message;
+  for (let i = 0; i <= splittedPath.length - 1; i++) {
+    const subPath = splittedPath[i];
+    const isLastElement = i >= splittedPath.length - 1;
+    if (subPath === "[]" && !isLastElement) continue;
+
+    value = getValue(subPath, value);
```

Comment:
I have a hard time understand the logic here, why looping looping through the each subPaths and override `value` ?

Reply:
As described in the comment above this method, you're supposed to do basically a slighlty modified version of the [lodash.get](https://lodash.com/docs/4.17.15#get) method.
So base on a string like `to.wallets.[]`, you're supposed to move inside of an object to find the value/values corresponding to that path.

So as an example:
```typescript
const obj = { 
  "to": {
    "wallets": [
      "0x123",
      "0x456"
    ]
  }
}

getValueFromPath("to.wallets.[]", obj) // ["0x123", "0x456"]
```

So the `getValue` is used to get the value of just 1 step:
```typescript
getValue("to", obj) // { wallets:  ["0x123", "0x456"] }
```

The loop contained in `getValueFromPath` is then repeating every step composed by the inital path (`"to.wallets.[]"`) in order to get the final value contained at the end of the path.

```typescript
getValueFromPath("to.wallets.[]", obj);

value = getValue("to", obj); // { wallets:  ["0x123", "0x456"] }
value = getValue("wallet", value /* { wallets:  ["0x123", "0x456"] } */ ) // ["0x123", "0x456"]
value = getValue("[]", value /* ["0x123", "0x456"] */ ) // ["0x123", "0x456"]
// because it's the last element but we could've continued if it was 
// an array of objects for example and the path was longer
```